collector/cpu_freebsd.go

```diff
@@ -74,50 +97,39 @@ func NewStatCollector() (Collector, error) {
 	}, nil
 }
 
-// Expose CPU stats using KVM.
+// Expose CPU stats using sysctl
 func (c *statCollector) Update(ch chan<- prometheus.Metric) (err error) {
-	if os.Geteuid() != 0 && os.Getegid() != 2 {
-		return errors.New("caller should be either root user or kmem group to access /dev/mem")
-	}
 
-	var errbuf *C.char
-	kd := C.kvm_open(nil, nil, nil, C.O_RDONLY, errbuf)
-	if errbuf != nil {
-		return errors.New("failed to call kvm_open()")
-	}
-	defer C.kvm_close(kd)
-
-	// The cp_time variable is an array of CPUSTATES long integers -- in
-	// the same format as the kern.cp_time sysctl.  According to the
-	// comments in sys/kern/kern_clock.c, the frequency of this timer will
-	// be stathz (or hz, if stathz is zero).
-	clockrate, err := getClockRate()
-	if err != nil {
-		return err
+	// We want time spent per-cpu per CPUSTATE
+	// CPUSTATES (number of CPUSTATES) is defined as 5U
+	// Order: CP_USER | CP_NICE | CP_SYS | CP_IDLE | CP_INTR
+	// sysctl kern.cp_times provides hw.ncpu * CPUSTATES long integers:
+	//   hw.ncpu * (space-separated list of the above variables)
+	//
+	// Each long integer is a counter incremented at frequency kern.clockrate.(stathz | hz)
+	//
+	// Look into sys/kern/kern_clock.c for details.
+
+	var ncpu C.int
+	var cpu_times_c_arr *C.double
```

Comment:
All names we control should follow the camelCase convention. Like `cpuTimesC` or something like that.